### PR TITLE
Optionally allow the system installation of MiniUPnP to be used

### PR DESCRIPTION
**Pull request by chewi**
## Description
As mentioned, I am packaging Sunshine for Gentoo Linux. Don't worry, we are not super strict about unbundling dependencies, and I understand that several are not feasible or worth the effort. This is one I would like to address though.

## Type of Change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Dependency update (updates to dependencies)
- [ ] Documentation update (changes to documentation)
- [ ] Repository update (changes to repository files, e.g. `.github/...`)

## Checklist
- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have added or updated the in code docstring/documentation-blocks for new or existing methods/components

## Branch Updates
LizardByte requires that branches be up-to-date before merging. This means that after any PR is merged, this branch
must be updated before it can be merged. You must also
[Allow edits from maintainers](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork).
- [X] I want maintainers to keep my branch updated

### CODE REVIEW

**Review by ReenigneArcher — CHANGES_REQUESTED** [review]
I don't have an issue with this PR.

My only concern is the Gentoo package deviating from what we provide, and users coming to us for support without sufficient details, thus wasting helpers time.

File: cmake/dependencies/common.cmake [hunk 1]
```diff
@@ -9,13 +9,32 @@ add_subdirectory(third-party/moonlight-common-c/enet)
...
+if(SUNSHINE_SYSTEM_MINIUPNP)
+    pkg_check_modules(MINIUPNP miniupnpc REQUIRED)
+    # Use includedir pkg-config variable rather than MINIUPNP_INCLUDE_DIRS
+    # defined above. The latter may be blank, as pkg-config sometimes omits -I
+    # flags for directories that are searched by default (e.g. /usr/include),
+    # but we need a value to append /miniupnpc to. Normally source files would
+    # prefix their #include directives with miniupnpc/, but this does not align
+    # with the directory structure of the git submodule used below.
+    pkg_get_variable(MINIUPNP_INCLUDE_DIR miniupnpc includedir)
+    include_directories(SYSTEM ${MINIUPNP_INCLUDE_DIR}/miniupnpc)
```
[the remark this answers is not in the document]
Comment (chewi) [reply cmake/dependencies/common.cmake:28]: You know I nearly added a line there. :laughing:

**Comment by chewi** [comment]
> My only concern is the Gentoo package deviating from what we provide, and users coming to us for support without sufficient details, thus wasting helpers time.

I do understand that concern. If it helps, I can add a notice to the package to say that users should report issues to us first.

**Comment by ReenigneArcher** [comment]
> If it helps, I can add a notice to the package to say that users should report issues to us first.

That would be appreciated!

**Review by ReenigneArcher — APPROVED** [review]

**Pull request merged**